File: lib/domain/usecases/get_assessment_scales.dart
import '../entities/assessment_scale.dart';
import '../repositories/assessment_repository.dart';

class GetAssessmentScales {
  const GetAssessmentScales(this._repository);

  final AssessmentRepository _repository;

  Future<List<AssessmentScale>> call() => _repository.getAssessmentScales();
}

File: src/application/useCases/getAssessmentScales.ts
import { AssessmentRepository } from '@domain/repositories/assessmentRepository';
import { AssessmentScale } from '@domain/entities/assessmentScale';

export const getAssessmentScales = async (
  repository: AssessmentRepository
): Promise<AssessmentScale[]> => {
  return repository.getScales();
};
